distributed/client.py

```diff
@@ -4638,7 +4635,12 @@ class will be instantiated with any extra keyword arguments.
         distributed.WorkerPlugin
         unregister_worker_plugin
         """
-        if isinstance(plugin, type):
+        if isinstance(plugin, type) or kwargs:
+            warnings.warn(
+                "passing plugin classes or kwargs is deprecated; "
+                "pass a constructed plugin instance instead",
+                category=FutureWarning,
+            )
```

Comment:
this PR also changes the behaviour to *always* bind the kwargs to the plugin if any are passed, if that's not desirable something like this could be done instead:

```suggestion
        if isinstance(plugin, type) or kwargs:
            warnings.warn(
                "passing plugin classes or kwargs is deprecated; "
                "pass a constructed plugin instance instead",
                category=FutureWarning,
            )
        if isinstance(plugin, type):
```